x-pack/filebeat/input/awss3/s3_objects.go

```diff
@@ -68,7 +70,7 @@ var errS3DownloadFailed = errors.New("S3 download failure")
 func newS3ObjectProcessorFactory(metrics *inputMetrics, s3 s3API, sel []fileSelectorConfig, backupConfig backupConfig) *s3ObjectProcessorFactory {
 	if metrics == nil {
 		// Metrics are optional. Initialize a stub.
-		metrics = newInputMetrics("", nil, 0)
+		metrics = newInputMetrics(v2.Context{MetricsRegistry: monitoring.NewRegistry()}, 0)
```

Comment:
Same question here and elsewhere, the registry is already provided by the context which is available to the caller of this function, shouldn't we use that instead of a new standalone registry in a faked context?